pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -276,9 +298,13 @@ public String getTaskType() {
           continue;
         }
 
-        Long prevWatermarkMs = mergeRollupTaskMetadata.getWatermarkMap().put(mergeLevel, windowStartMs);
+        Long prevWatermarkMs = mergeRollupTaskMetadata.getWatermarkMap().get(mergeLevel);
+        mergeRollupTaskMetadata.getWatermarkMap().put(mergeLevel, windowStartMs);
```

Comment:
Please keep the existing code because it gives the same result but more efficient

Reply:
fixed